### ISSUE TITLE
[RFC] Provides all model in the hub as backbone in a consistent way.

### ISSUE THREAD

**Issue by tchaton** [facebookresearch/pytorchvideo#115]
## 🚀 Feature

Currently, Lightning Flash integrates PyTorchVideo to provides simple access to the VideoClassifier.

However, the model definition through the hub aren't consistent.

The new MultiScale VIT has a different structure and provide a different compared past models.

Here is how we are currently accessing the num_classes. It is also hard to remove the head to finetune a model on new data.

```py
            try:
                num_features = self.backbone.blocks[-1].proj.out_features
            except:
                num_features = self.backbone.head.proj.out_features
```

The proposal would be to create model in a consistent way where both backbone and head are isolated and can easily be swapped in required.

```py
def mvit_base_16(
    num_classes: Optional[int] = None
):

    backbone = MVITBackbone(pretrained=True)
    head = MVITHead(num_classes=num_classes)
    if pretrained:
        backbone.load...
    return Classifier(backbone, head)

model = mvit_base_16()
model.num_features # backbone output_dim
model.num_classes # head output_dim
```

Furthermore, the transforms used during pre-training of the models aren't paired with the models and this would be a great addition.

NOTE: Please look at the existing list of Issues tagged with the label ['enhancement`](https://github.com/facebookresearch/pytorchvideo/issues?q=label%3Aenhancement). **Only open a new issue if you do not see your feature request there**.

## Pitch

NOTE: we only consider adding new features if they are useful for many users.
